File: proto_ver2/python/reader.py



"""
Protocol buffer reader in Python

author: Atsushi Sakai
"""

import addressbook_pb2


def main():
    """
    main func
    """
    print("start!!")

    address_book = addressbook_pb2.AddressBook()

    # Read the existing address book.
    file_path = open("pbdata_py.dat", "rb")  # read
    # file_path = open("../../sample1/python/pbdata_py.dat",
    # "rb")  # read old data

    address_book.ParseFromString(file_path.read())
    file_path.close()

    print(address_book)

    print("done!!")


if __name__ == '__main__':
    main()

File: proto_ver1/java/src/main/java/reader/Reader.java
package reader;

import java.io.FileInputStream;

import tutorial.Addressbook.AddressBook;

public class Reader {

    public static void main(String[] args) throws Exception {
        System.out.println("Protocol buffer reader start!!");

        AddressBook addressBook =
            AddressBook.parseFrom(new FileInputStream("pbdata_java.dat"));

        System.out.println(addressBook); // Human readable print

        System.out.println("Protocol buffer reader done!!");
    }

}
